src/webgpu/util/floating_point.ts

```diff
@@ -0,0 +1,3090 @@
+    ),
+    zeroVector: {
+      2: [kF32ZeroInterval, kF32ZeroInterval],
+      3: [kF32ZeroInterval, kF32ZeroInterval, kF32ZeroInterval],
+      4: [kF32ZeroInterval, kF32ZeroInterval, kF32ZeroInterval, kF32ZeroInterval],
+    },
+    anyVector: {
+      2: [kF32AnyInterval, kF32AnyInterval],
+      3: [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+      4: [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+    },
+    anyMatrix: {
+      2: {
+        2: [
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+        ],
+        3: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+        4: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+      },
+      3: {
+        2: [
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+        ],
+        3: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+        4: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+      },
+      4: {
+        2: [
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval],
+        ],
+        3: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+        4: [
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+          [kF32AnyInterval, kF32AnyInterval, kF32AnyInterval, kF32AnyInterval],
+        ],
+      },
+    },
+  };
+
+  public constructor() {
+    super('f32');
+  }
+
+  public constants(): FPConstants {
+    return F32Context._constants;
+  }
+
+  // Overrides - Utilities
+  public readonly correctlyRounded = correctlyRoundedF32;
+  public readonly isFinite = isFiniteF32;
+  public readonly isSubnormal = isSubnormalNumberF32;
+  public readonly flushSubnormal = flushSubnormalNumberF32;
+  public readonly oneULP = oneULPF32;
+  public readonly scalarBuilder = f32;
+}
+
+export const kFPContext = {
```

Comment:
This is going to get used a lot. I'd be inclined to just call it `FP` to keep the noise down on usage. Or even just expose `f32` without wrapping it in a object.

Reply:
I have renamed it FP. I prefer having it in const object like this, for the future where there is more than one kind defined, so code can use the kind as a key `FP[kind]`

Reply:
Ack